### PR TITLE
aalc: add sensor level 3 and event log

### PR DESCRIPTION
**Pull request by HungYi-Li**
Summary:
- modify fan table
- block auto tune command util bic is ready
- disable auto tune will not set duty to 0
- add more error log event
- add sensor COOLING_CAPACITY
- add sensor BPB_RACK_LEVEL_3
- extend COOLING_CAPACITY modbus command to 2 register
- add modbus shell command for test
- modify the threshold of y_filter to be dynamic
- fix threshold struct member last_status to the correct size

Test Plan:
- Build code: PASS

### CODE REVIEW

File: meta-facebook/aalc-rpu/src/platform/plat_fsc.h [hunk 1]
```diff
@@ -46,10 +46,11 @@ enum FSC_TYPE {
 	FSC_TYPE_DEFAULT,
 };
 
+/*
```
Comment (waffle2k) [inline meta-facebook/aalc-rpu/src/platform/plat_fsc.h:49]: pls delete commented code

File: meta-facebook/aalc-rpu/src/platform/plat_fsc.c [hunk 2]
```diff
@@ -159,6 +160,7 @@ static uint8_t calculatePID(zone_cfg *zone_p, uint8_t *duty)
 
 		int16_t temp = (int16_t)tmp;
 
+		/*
```
Comment (waffle2k) [inline meta-facebook/aalc-rpu/src/platform/plat_fsc.c:163]: pls delete commented code

File: meta-facebook/aalc-rpu/src/platform/plat_fsc.c [hunk 3]
```diff
@@ -22,6 +22,7 @@ static uint8_t fsc_poll_flag = 1;
 extern zone_cfg zone_table[];
 extern uint32_t zone_table_size;
 
+/*
```
Comment (waffle2k) [inline meta-facebook/aalc-rpu/src/platform/plat_fsc.c:25]: pls delete commented code

File: meta-facebook/aalc-rpu/src/platform/plat_hwmon.c [hunk 4]
```diff
@@ -415,7 +418,13 @@ uint8_t pwm_control(uint8_t group, uint8_t duty)
...
 	if (!get_status_flag(STATUS_FLAG_AUTO_TUNE)) {
 		auto_tune_time = k_uptime_get();
-		set_pwm_group(group, 0);
+		return 0;
+	}
+
+	// keep 0 duty when first enabling auto tune
+	static bool is_init = false;
+	if (!is_init) {
+		is_init = true;
```
Comment (waffle2k) [inline meta-facebook/aalc-rpu/src/platform/plat_hwmon.c:427]: does it make more sense to move this down to line 439 once we know that things are initialized?

File: meta-facebook/aalc-rpu/src/platform/plat_hwmon.c [hunk 5]
```diff
@@ -415,7 +418,13 @@ uint8_t pwm_control(uint8_t group, uint8_t duty)
 	static int64_t auto_tune_time;
 	if (!get_status_flag(STATUS_FLAG_AUTO_TUNE)) {
 		auto_tune_time = k_uptime_get();
-		set_pwm_group(group, 0);
+		return 0;
+	}
+
+	// keep 0 duty when first enabling auto tune
```
Comment (waffle2k) [inline meta-facebook/aalc-rpu/src/platform/plat_hwmon.c:424]: could you please expand on the comment to describe why it's required that we return early on first call?

**Review by waffle2k — CHANGES_REQUESTED** [review]
overall good, just some minor nits on commenting and the logic around the early initialization exit

File: meta-facebook/aalc-rpu/src/platform/plat_fsc.h (hunk 1 above)
Comment (HungYi-Li) [reply meta-facebook/aalc-rpu/src/platform/plat_fsc.h]: delete in 32fa14b

File: meta-facebook/aalc-rpu/src/platform/plat_fsc.c (hunk 2 above)
Comment (HungYi-Li) [reply meta-facebook/aalc-rpu/src/platform/plat_fsc.c]: delete in https://github.com/facebook/OpenBIC/commit/32fa14b03a5e6c38582ff51d0685ecefc98cf37f

File: meta-facebook/aalc-rpu/src/platform/plat_fsc.c (hunk 3 above)
Comment (HungYi-Li) [reply meta-facebook/aalc-rpu/src/platform/plat_fsc.c]: delete in https://github.com/facebook/OpenBIC/commit/32fa14b03a5e6c38582ff51d0685ecefc98cf37f

File: meta-facebook/aalc-rpu/src/platform/plat_hwmon.c (hunk 5 above)
Comment (HungYi-Li) [reply meta-facebook/aalc-rpu/src/platform/plat_hwmon.c:424]: The current design is that when the AC is on, the pump will remain at duty 0 until someone turns on auto tune.

File: meta-facebook/aalc-rpu/src/platform/plat_hwmon.c (hunk 4 above)
Comment (HungYi-Li) [reply meta-facebook/aalc-rpu/src/platform/plat_hwmon.c:427]: Based on the reply of the next commit, if the duty is set after line 439, the pump will start when the system is ready

**Pull request closed without merging**